private/check/check.go

```diff
@@ -20,16 +21,21 @@ import (
 type TestFixtureGetter struct {
 	fsys  fs.FS
 	cache map[string]*verify.TestFixture
+	mut   sync.RWMutex
 }
 
 func NewTestFixtureGetter(fsys fs.FS) *TestFixtureGetter {
 	return &TestFixtureGetter{
 		fsys:  fsys,
 		cache: make(map[string]*verify.TestFixture),
+		mut:   sync.RWMutex{},
 	}
 }
 
-func (g *TestFixtureGetter) LoadTestFixture(path string) (*verify.TestFixture, error) {
+func (g *TestFixtureGetter) Load(path string) (*verify.TestFixture, error) {
+	g.mut.Lock()
+	defer g.mut.Unlock()
```

Comment:
Shouldn't we:
1. Acquire a read lock first to check whether the item exists in the cache
2. Acquire a write lock, check  whether the item exists again
3. Load the item and release the write lock

Reply:
Ha, I thought that was overkill for this scenario. But it's certainly more optimal. Will update

Reply:
Is it likely to be accessed by multiple goroutines anyway? I can't imagine that being the case and we could potentially get rid of the mutex completely.

Reply:
No, you're right. In the `spitfire` usage the writes and reads happen sequentially. I'll do away with the mutex